<launch>
    <!-- <node name="usb_cam" pkg="usb_cam" type="usb_cam_node" output="screen" >
        <param name="video_device" value="/dev/video0" />
        <param name="image_width" value="640" />
        <param name="image_height" value="480" />
        <param name="pixel_format" value="yuyv" />
        <param name="framerate" value="2" />
        <param name="color_format" value="yuv422p" />
        <param name="camera_frame_id" value="usb_cam" />
        <param name="io_method" value="mmap"/>
    </node> -->
    <!-- This node displays the image from the specified topic, it is here for testing purposes -->
    <!--node name="image_view" pkg="image_view" type="image_view" respawn="false" output="screen">
        <remap from="image" to="/usb_cam/image_raw"/>
        <param name="autosize" value="true" />
    </node-->

    <node pkg="person_recognition" type="unsw_person_recognition_node.py" name="unsw_person_recognition_node" respawn="False" output="screen" >
        <!--param name="input_topic" value="/hsrb/head_center_camera/image_raw" /-->
        <!--param name="input_topic" value="/hsrb/head_center_camera/image_compressed" /-->
        <param name="input_topic" value="/unsw_vision/detections/objects/positions" />
        <!-- <param name="input_topic" value="/unsw_vision/detections/objects" /> -->
        <param name="output_topic" value="/unsw_vision/detections/persons" />
        <param name="vis_output_topic" value="/unsw_vision/detections/final/visualization3" />
        <param name="use_world_model" value="False"/>
        <param name="display_img" value="False" /> 
        <param name="visualize" value="True"/>        
        <param name="save_img" value="False"/> 
        <param name="img_path" value="$(env UNSW_WS)/VISION/person_recognition/images/"/>
        <param name="db_ini" value="$(env UNSW_WS)/WORLD_MODEL/world_model/world_model_kb/scripts/database.ini"/>
        <param name="queue_size" value="100" />
        <param name="use_gpu" value="True"/>
        <param name="scale" value="1.0" />
        <param name="face_threshold" value="0.5" />
        <!-- THIS IS DISTANCE THRESHOLD so it will be match if below this value. Increase the value if there are too many agents being created -->
        <param name="feature_threshold" value="0.230" />
    </node> 

    <!-- <node pkg="person_recognition" type="pr_timer_node.py" name="pr_timer_node" respawn="False" output="screen" >      
        <param name="db_ini" value="$(env UNSW_WS)/world_model/world_model_kb/scripts/database.ini"/>
        <param name="timer_interval" value="60.0" />    time in seconds -->
    <!-- </node> -->

</launch>
